dhcp-protocol/src/main/java/org/anarres/dhcp/v6/options/Dhcp6Options.java

```diff
@@ -69,21 +93,25 @@ public void addAll(@CheckForNull Iterable<? extends Dhcp6Option> options) {
 
     /**
      * Remove instances of the given option class.
-     * 
+     *
      * @param type
      */
     public void remove(@Nonnull Class<? extends Dhcp6Option> type) {
         Dhcp6OptionsRegistry registry = Dhcp6OptionsRegistry.getInstance();
-        remove(registry.getOptionTag(type));
+        removeAll(registry.getOptionTag(type));
     }
 
-    /**
-     * Remove options matching the given tag
-     * 
-     * @param tag
-     */
-    public void remove(short tag) {
-        options.remove(tag);
+    public Collection<Dhcp6Option> removeAll(final short optionTag) {
+        return options.removeAll(optionTag);
+    }
+
+    public void remove(@Nonnull Class<? extends Dhcp6Option> type, Dhcp6Option value) {
+        Dhcp6OptionsRegistry registry = Dhcp6OptionsRegistry.getInstance();
+        remove(value, registry.getOptionTag(type));
+    }
+
+    public boolean remove(final Dhcp6Option value, final short optionTag) {
```

Comment:
This method has inconsistent argument order to the others.

Also, if we're removing a value, we can derive the tag/type from the value, so we don't actually require the `Class, Option` or `short, Option` variants?